cloud/aws/services/elb/loadbalancer.go

```diff
@@ -0,0 +1,183 @@
+			UnhealthyThreshold: 3,
+		},
+		SecurityGroupIDs: []string{network.SecurityGroups[v1alpha1.SecurityGroupControlPlane].ID},
+		Tags: map[string]string{
+			fmt.Sprintf("%s%s", TagNameKubernetesClusterPrefix, clusterName): string(ResourceLifecycleOwned),
+			TagNameAWSClusterAPIRole: "apiserver",
+		},
+	}
+
+	for _, sn := range network.Subnets.FilterPrivate() {
+		res.SubnetIDs = append(res.SubnetIDs, sn.ID)
+	}
+
+	return res
+}
+
+func (s *Service) createClassicELB(spec *v1alpha1.ClassicELB) (*v1alpha1.ClassicELB, error) {
+	input := &elb.CreateLoadBalancerInput{
+		LoadBalancerName: aws.String(spec.Name),
+		Subnets:          aws.StringSlice(spec.SubnetIDs),
+		SecurityGroups:   aws.StringSlice(spec.SecurityGroupIDs),
+		Scheme:           aws.String(string(spec.Scheme)),
+	}
+
+	for key, value := range spec.Tags {
+		input.Tags = append(input.Tags, &elb.Tag{Key: aws.String(key), Value: aws.String(value)})
+	}
+
+	for _, ln := range spec.Listeners {
+		input.Listeners = append(input.Listeners, &elb.Listener{
+			Protocol:         aws.String(string(ln.Protocol)),
+			LoadBalancerPort: aws.Int64(ln.Port),
+			InstanceProtocol: aws.String(string(ln.InstanceProtocol)),
+			InstancePort:     aws.Int64(ln.InstancePort),
+		})
+	}
+
+	out, err := s.ELB.CreateLoadBalancer(input)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create classic load balancer: %v", spec)
+	}
+
+	if spec.HealthCheck != nil {
+		hc := &elb.ConfigureHealthCheckInput{
+			LoadBalancerName: aws.String(spec.Name),
+			HealthCheck: &elb.HealthCheck{
+				Target:             aws.String(spec.HealthCheck.Target),
+				Interval:           aws.Int64(int64(spec.HealthCheck.Interval.Seconds())),
+				Timeout:            aws.Int64(int64(spec.HealthCheck.Timeout.Seconds())),
+				HealthyThreshold:   aws.Int64(spec.HealthCheck.HealthyThreshold),
+				UnhealthyThreshold: aws.Int64(spec.HealthCheck.UnhealthyThreshold),
+			},
+		}
+
+		if _, err := s.ELB.ConfigureHealthCheck(hc); err != nil {
+			return nil, errors.Wrapf(err, "failed to configure health check for classic load balancer: %v", spec)
+		}
+	}
+
+	res := spec.DeepCopy()
+	res.DNSName = *out.DNSName
+	return res, nil
+}
+
+func (s *Service) describeClassicELB(name string) (*v1alpha1.ClassicELB, error) {
+	input := &elb.DescribeLoadBalancersInput{
+		LoadBalancerNames: []*string{aws.String(name)},
+	}
+
+	out, err := s.ELB.DescribeLoadBalancers(input)
+	if (err != nil && strings.Contains(err.Error(), "There is no ACTIVE Load Balancer")) || (out != nil && len(out.LoadBalancerDescriptions) == 0) {
+		return nil, NewNotFound(errors.Errorf("no classic load balancer found with name %q", name))
+	} else if err != nil {
+		return nil, errors.Wrapf(err, "failed to describe classic load balancer: %s", name)
+	}
+
+	return fromSDKTypeToClassicELB(out.LoadBalancerDescriptions[0]), nil
+}
+
+func fromSDKTypeToClassicELB(v *elb.LoadBalancerDescription) *v1alpha1.ClassicELB {
+	lb := &v1alpha1.ClassicELB{
```

Comment:
is there any reason this isn't a direct return? if it's style preference please leave it, i'm just curious